lametro/views.py

```diff
@@ -811,3 +815,119 @@ def refresh_guid_trigger(request, refresh_key):
     except AttributeError:
         print('You need a refresh_key in your local deployment settings files to access this.')
     return HttpResponse(403)
+
+
+class SmartLogicAPI(ListView):
+    api_key = SMART_LOGIC_KEY
+    query_format = 'https://cloud.smartlogic.com/svc/' \
+        + SMART_LOGIC_ENVIRONMENT + '/ses/CombinedModel/hints/{query}.json'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.token_refreshed = False
+
+    @property
+    def access_token(self):
+        session = self.request.session
+
+        if not session.get('smart_logic_token'):
+            session['smart_logic_token'] = self._generate_token()
+
+        return session['smart_logic_token']
+
+    def render_to_response(self, context):
+        '''
+        Return response as JSON.
+        '''
+        try:
+            response = self.get_queryset(context)
+            return JsonResponse(response)
+        except json.JSONDecodeError:
+            return JsonResponse({ 'status': 'false', 'message': 'No topic found' }, status=500)
+
+
+    def get_queryset(self, *args, **kwargs):
+        '''
+        Hit the SmartLogic endpoint. Tokens expire in two weeks, so if we get
+        an authentication-related status code, refresh the token and try again
+        once before failing out.
+        '''
+        url = self.query_format.format(query=self.request.GET['query'])
+        headers = {'Authorization': 'Bearer ' + self.access_token}
+
+        try:
+            response = requests.get(url, headers=headers)
+        except HTTPError as e:
+            print('Could not communicate with SmartLogic: {}'.format(e))
+        except Exception:
+            raise
+        else:
+            if response.status_code == 200:
```

Comment:
The issue here was when searching for "con," there was a 200 response from SmartLogic—it was the HTML of an error page. We handle the `json.JSONDecodeError` that causes here.

Reply:
nice catch!

Reply:
can you leave a comment to that effect in the except block?